plugins/modules/win_gpo_manage.ps1

```diff
@@ -0,0 +1,147 @@
+                if(!($gpos -contains $Name)) {
+                    #if no GPO exist
+                    New-GPO -Name $Name -WhatIf:$check_mode
+                    $module.result.changed = $true
+                }
+                Import-GPO -TargetName $Name -Path $Path -BackupId $ID.Name -WhatIf:$check_mode
+                $module.result.imported.Add($Name)
+                $module.result.changed = $true
+            }
+        }
+    } else {
+        $module.FailJson("Folder has no Sub Folders for import ")
+    }
+}
+function Export-GPOs {
+    param (
+        $ExportFolder
+    )
+    if((Test-Path -LiteralPath $folderpath) -and !($override)) {
+        #Folder Exist and override is false
+        $module.FailJson("folder $folderpath exists already")
+    }
+    $gpo = Get-GPO -All
+    foreach ($Entry in $gpo) {
+        $Path = $ExportFolder + "\" + $entry.Displayname
+        if((Test-Path -LiteralPath $Path) -and !($override)) {
+            $module.FailJson("Folder already exists, Override is false ")
+        } else {
+            if(!(Test-Path -LiteralPath $Path)) {
+                #Folder doesnt exist
+                New-Item -ItemType directory -Path $Path -WhatIf:$check_mode
+            }
+            if((Test-Path -LiteralPath $Path) -and ($override)) {
+                Remove-Item -ItemType directory -Path $Path -WhatIf:$check_mode
+                New-Item -ItemType directory -Path $Path -WhatIf:$check_mode
+            }
+            Backup-GPO -Guid $Entry.id -Path $Path -WhatIf:$check_mode
+            $module.result.exported.Add($entry.Displayname)
+        }
+    }
+}
+if ($diff_mode) {
+    $module.result.diff = @{}
+}
+if ($gpomode -eq "remove") {
+    #Removes GPOs
+    $gpo = (Get-GPO -All).Displayname
+    $module.diff.before = $gpo | Out-String
+    if ($removedgpo.count -gt 0) {
+        foreach ($gporemove in $removedgpo) {
+            if($gpos -contains $removedgpo) {
+                #check if GPO exist
+                try {
+                    Remove-GPO -Name $gporemove -WhatIf:$check_mode
+                    $module.result.removed.Add($gporemove)
+                    $module.result.changed = $true
+                } catch {
+                    $module.FailJson("an exception occurred when deleting GPO - $($_.Exception.Message)")
+                }
+            } else {
+                $module.result.changed = $false #GPO does not exist - Skipping that Part
+            }
+        }
+    } else {
+        $module.result.changed = $false #there are no gpos
+    }
+    if ($diff_mode -and $result.changed) {
+        if (!$check_mode) {
+            #only check
+            $module.result.diff.after = [Array]$removedgpo | Out-String
+        } else {
+            #remove gpos
+            $gpoafter = (Get-GPO -All).Displayname
+            $after = [System.Collections.Generic.List`1[String]]
+            $after.AddRange($gpoafter)
+            $module.result.diff.after = $after | Out-String
```

Comment:
`$after` is now a type and not the list you want it to be so running `$after.AddRange($gpoafter)` will not work. Ultimately you can simplify this to just

```suggestion
            $module.Diff.after = @((Get-Gpo -All).Displayname)`
```